### PR TITLE
Initial payment controller

### PR DESCRIPTION
**Pull request by rizperdana**
### What this PR does (required):
- This PR will add payment functionality on backend
- Current payment controller will support to fetch all payment or certain payment by id
- Also we can pay or cancel our payment

### Any information needed to test this feature (required):
- You can check current route on:
1. `http://localhost:3001/payments` for Get All Payments
2. `http://localhost:3001/payments/:id` Get single Payments
3. `http://localhost:3001/payments/:id/pay` Make Payment with Put Method
4. `http://localhost:3001/payments/:Id/cancel` Cancel Payment with Put Method
- Havent create any payment test case or seeders here

### CODE REVIEW

File: server/controllers/payment.js [hunk 1]
```diff
@@ -0,0 +1,84 @@
+const asyncHandler = require("express-async-handler");
+const Payment = require("../models/Payment");
+
+// @route GET /payments
+// @desc Get all payment from user
+// @access Private
```
Comment (sundayezeilo) [inline server/controllers/payment.js]: Good documentation

File: server/controllers/payment.js [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+    });
+  } else {
+    res.status(404);
+    throw new Error("Pls enter valid payment id");
+  }
+});
+
+// @route PUT /payments/:id/pay
+// @desc Pay current unpaid payment
+// @access Private
```
Comment (sundayezeilo) [inline server/controllers/payment.js]: :+1:

**Review by sundayezeilo — COMMENTED** [review]
Routes and controllers look good! But always link your ticket in your PRs.

File: server/controllers/payment.js [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+exports.cancelPayment = asyncHandler(async (req, res, next) => {
+  const userId = req.user.id;
+  const { id } = req.params;
+  const merchantPayment = await Payment.findOne({ id: id, sitterId: userId });
+  if (!merchantPayment) {
+    res.status(203);
+    throw new Error("Only Merchant Can Cancel");
+  } else {
+    merchantPayment.set({ paid: false });
```
Comment (moffatethan) [inline server/controllers/payment.js]: We should also add a property on to the payment schema. `cancel` and set that to true here.

**Review by moffatethan — CHANGES_REQUESTED** [review]
Super close - just one comment. 😄

File: server/controllers/payment.js (hunk 3 above)
Comment (rizperdana) [reply server/controllers/payment.js]: Added

File: server/controllers/notifications.js [hunk 4]
```diff
@@ -64,7 +64,11 @@ exports.createNotification = asyncHandler(async (req, res, next) => {
 // @access Private
 exports.markNotificationAsRead = asyncHandler(async (req, res, next) => {
   const query = { _id: req.body.id };
-  const notification = await Notification.findOneAndUpdate(query, { read: true }, { new: true });
+  const notification = await Notification.findOneAndUpdate(
+    query,
+    { read: true },
+    { new: true }
+  );
```
Comment (moffatethan) [inline server/controllers/notifications.js]: This actually shouldn't be here in this file

**Review by moffatethan — APPROVED** [review]

**Pull request merged**